discord_bot/discord_bot.py

```diff
@@ -144,24 +142,45 @@ def load_config(config_file_path):
                     self.logger.warning("Invalid Session")
                 elif message["op"] == Opcodes.DISPATCH:
                     event = message["t"]
-                    if event == Events.READY:
+                    if event == Events.READY.value:
```

Comment:
Doesn't this already compare by value? The purpose of an enum is to make it as easy as possible to compare values intuitively without adding additional syntax, so we should probably remove the `.value` from all if cases.

Reply:
Going through the debugger. 
I found that Events.READY actually don't equal 'READY'.
https://docs.python.org/3/library/enum.html
8.13.14.1.3. Using a descriptive string
This example is why I had to use .value for all cases.

Reply:
See https://docs.python.org/3/library/enum.html#others

Time to use a little mix-in magic to make it work, but basically your issue is that the way it compares equals (I believe the hidden method is called `__eq__`) is not using the `str` method. If you use a `str` mix-in then it should work properly.